### PR TITLE
Enumeration check

### PR DESCRIPTION
**Pull request by adjivas**
Check case for #5 .

* [x] undefined_field_antisymmetric: A antisymmetric should refers a defined fields from same enum.

### CODE REVIEW

**Comment by adjivas** [comment]
This PR is ready for a review.

File: telamon-gen/src/ast/choice/enumeration.rs [hunk 1]
```diff
@@ -10,64 +10,180 @@ pub struct EnumDef {
...
                     }
                 },
+            }
+
+        }
+        Ok(())
+    }
+
+    /// This checks that there isn't any doublon in parameter list.
+    fn check_redefinition_parameter(&self) -> Result<(), TypeError> {
```
Comment (ulysseB) [inline telamon-gen/src/ast/choice/enumeration.rs]: This should be a method common to all `ChoiceDef`.

File: telamon-gen/src/ast/mod.rs [hunk 2]
```diff
@@ -67,9 +67,11 @@ pub enum TypeError {
     /// Undefinition of set, enum or field.
     Undefined(Spanned<String>),
     /// Unvalid arguments of a symmetric enum.
-    BadSymmetricArg(Vec<VarDef>),
+    BadSymmetricArg(Spanned<String>, Vec<VarDef>),
```
Comment (ulysseB) [inline telamon-gen/src/ast/mod.rs]: Add a comment to explain what is the `Spanned<String>` or use named fields.

File: telamon-gen/src/ast/choice/enumeration.rs (hunk 1 above)
Comment (adjivas) [reply telamon-gen/src/ast/choice/enumeration.rs]: That solved by the [PR 94](https://github.com/ulysseB/telamon/pull/94)?

File: telamon-gen/src/ast/mod.rs (hunk 2 above)
Comment (adjivas) [reply telamon-gen/src/ast/mod.rs]: Solved with a Alias _SpannedName_ and _Name_

File: telamon-gen/src/ast/choice/enumeration.rs (hunk 1 above)
Comment (ulysseB) [reply telamon-gen/src/ast/choice/enumeration.rs]: Ok, I will look for this in #94

File: telamon-gen/src/ast/mod.rs (hunk 2 above)
Comment (ulysseB) [reply telamon-gen/src/ast/mod.rs]: Still not clear for me. I think the alias adds complexity but does not explains things. I would rather have a comment instead.

**Comment by ulysseB** [comment]
I think you misunderstood me. The problem was not in the structs (where the field name was already providing the info) but the the definition of the `TypeError` enum, specifically the fields of  `BadSymmetricArg(Spanned<String>, Vec<VarDef>),`. Either document the or given them a name.

**Comment by adjivas** [comment]
@ulysseB yes right, normally that fixed.
Ready for review again.

File: telamon-gen/src/ast/error.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#[derive(Debug, PartialEq)]
+pub enum TypeError {
+    /// Redefinition of a name and hint.
+    Redefinition { object_kind: Spanned<Hint>, object_name: Spanned<String> },
+    /// Undefinition of set, enum or field.
+    Undefined { object_name: Spanned<String> },
+    /// Unvalid arguments of a symmetric enum.
+    BadSymmetricArg { object_name: Spanned<String>, object_variables: Vec<VarDef> },
+    /// Missing
+    /// Append when the Set's object has a missing field.
```
Comment (ulysseB) [inline telamon-gen/src/ast/error.rs]: typo: Append -> Happens, idem for `Conflict`

**Pull request merged**